pkg/scanners/trivy/types.go

```diff
@@ -86,92 +85,100 @@ func DefaultConfig() *Config {
-		if err != nil {
-			log.Error(err, "could not find image by reference", "imageID", img.ImageID, "reference", ref)
-			cleanup()
-			continue
-		}
-		log.Info("found image with id under reference", "imageID", img.ImageID, "ref", ref)
-
-		disabledAnalyzers := appendDisabledAnalyzers(analyzer.TypeConfigFiles, analyzer.TypeLockfiles, analyzer.TypeIndividualPkgs, analyzer.TypeLanguages)
-
-		artifactToScan, err := artifactImage.NewArtifact(dockerImage, s.trivyScanConfig.fscache, artifact.Option{
-			Offline:           true,
-			DisabledAnalyzers: disabledAnalyzers,
-			DisabledHandlers:  []fanalTypes.HandlerType{fanalTypes.UnpackagedPostHandler, fanalTypes.MisconfPostHandler},
-			SBOMSources:       []string{},
-		})
-		if err != nil {
-			log.Error(err, "error registering config for artifact", "imageID", img.ImageID, "reference", ref)
-			cleanup()
+		stdout := new(bytes.Buffer)
+		stderr := new(bytes.Buffer)
+
+		cliArgs := s.config.cliArgs(refs[i])
+		cmd := exec.Command(trivyCommandName, cliArgs...)
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+
+		log.V(1).Info("scanning image ref", "ref", refs[i], "cli_invocation", fmt.Sprintf("%s %s", trivyCommandName, strings.Join(cliArgs, " ")))
+		if err := cmd.Run(); err != nil {
+			log.Error(err, "error scanning image", "imageID", img.ImageID, "reference", refs[i], "stderr", stderr.String())
 			continue
 		}
 
-		imageScanContext, cancel := context.WithTimeout(context.Background(), perImageTimeout)
-		defer cancel()
-
-		scanner := scanner.NewScanner(s.trivyScanConfig.localScanner, artifactToScan)
-		report, err := scanner.ScanArtifact(imageScanContext, s.trivyScanConfig.scanOptions)
-		if err != nil {
-			log.Error(err, "error scanning image", "imageID", img.ImageID, "reference", ref)
-			cleanup()
+		var report trivyTypes.Report
+		if err := json.Unmarshal(stdout.Bytes(), &report); err != nil {
+			log.Error(err, "error unmarshaling report", "imageID", img.ImageID, "reference", refs[i], "report", stdout.String(), "stderr", stderr.String())
 			continue
 		}
 
-		if s.userConfig.DeleteEOLImages {
+		if s.config.DeleteEOLImages {
 			if report.Metadata.OS != nil && report.Metadata.OS.Eosl {
-				log.Info("image is end of life", "imageID", img.ImageID, "reference", ref)
+				log.Info("image is end of life", "imageID", img.ImageID, "reference", refs[i])
 				return StatusNonCompliant, nil
 			}
 		}
 
-		for i := range report.Results {
-			for j := range report.Results[i].Vulnerabilities {
-				if s.userConfig.Vulnerabilities.IgnoreUnfixed && report.Results[i].Vulnerabilities[j].FixedVersion == "" {
-					continue
-				}
-
-				if report.Results[i].Vulnerabilities[j].Severity == "" {
-					report.Results[i].Vulnerabilities[j].Severity = severityUnknown
-				}
-
-				if severityMap[report.Results[i].Vulnerabilities[j].Severity] {
-					return StatusNonCompliant, nil
-				}
+		for j := range report.Results {
+			if len(report.Results[j].Vulnerabilities) > 0 {
+				// TODO: Surface vulnerability results
```

Comment:
i think we can remove this comment since there's an issue tracking this if there's interest